lib/iris/coord_systems.py

```diff
@@ -31,6 +31,34 @@
 import cartopy.crs as ccrs
 
 
+def _dict_nearly_equal(dict1, dict2):
+    # Allow coordinate system comparison (floating point numbers with
+    # tolerance).
+    result = True
+    if result:
+        for key, value in dict1.iteritems():
+            if key not in dict2:
+                result = False
+                break
+            if hasattr(dict2[key], '__dict__') and hasattr(value, '__dict__'):
+                result = _dict_nearly_equal(dict2[key].__dict__,
+                                            value.__dict__)
+            elif hasattr(dict2[key], '__dict__') != hasattr(value, '__dict__'):
+                result = False
+            else:
+                # We intentionally do not use a global tolerance for coordinate
+                # system comparison.
+                if True in [dict2[key] is None, value is None]:
+                    result = dict2[key] == value
+                elif np.isreal(dict2[key]) and np.isreal(value):
+                    result = abs(dict2[key] - value) < 1
```

Comment:
This is an intentionally naive/very accepting here.  Feel free to share ideas you might have (np.allclose etc.  what might fit more generally package wide??).